analytics.go

```diff
@@ -250,25 +251,25 @@ func (c *client) upload(b []byte) error {
 	}
 
 	defer res.Body.Close()
-	c.report(res)
-
-	return nil
+	return c.report(res)
 }
 
 // Report on response body.
-func (c *client) report(res *http.Response) {
+func (c *client) report(res *http.Response) (err error) {
+	var body []byte
+
 	if res.StatusCode < 400 {
```

Comment:
True, though I think we should still stick to normal HTTP semantics (2xx/3xx OK and 4xx, 5xx not ok). We've talked internally about changing the 200 OK to 202 Accepted and being too aggressive with the error codes might bite us later.